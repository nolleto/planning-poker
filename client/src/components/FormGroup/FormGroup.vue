<template>
  <div :class="bootstrap.formGroup">
    <label v-if="name" :for="name">
      {{ title }}
    </label>

    <slot></slot>
  </div>
</template>

<script>
import styles from '@/styles'

export default {
  name: 'FormGroup',

  props: {
    name: {
      type: String
    },

    title: {
      type: String
    }
  },

  computed: {
    bootstrap: () => styles.bootstrap
  }
}
</script>
