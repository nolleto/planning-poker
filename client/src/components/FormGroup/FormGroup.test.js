import FormGroup from './index'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(FormGroup)
}

describe('FormGroup component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    describe('When passing name and title props', () => {
      beforeAll(() => {
        wrapper.setProps({
          name: 'name',
          title: 'Name'
        })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
