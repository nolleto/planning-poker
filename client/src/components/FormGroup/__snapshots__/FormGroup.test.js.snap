// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup component Snapshots When passing name and title props matches 1`] = `
<div
  class="formGroup"
>
  <label
    for="name"
  >
    
    Name
  
  </label>
   
</div>
`;

exports[`FormGroup component Snapshots matches 1`] = `
<div
  class="formGroup"
>
  <!---->
   
</div>
`;
